src/composables/useTransactions.ts

```diff
@@ -0,0 +1,327 @@
+  from: string;
+  lastCheckedBlockNumber?: number;
+  status: TransactionStatus;
+};
+
+export type NewTransaction = Pick<
+  Transaction,
+  'id' | 'type' | 'summary' | 'receipt' | 'action'
+>;
+
+const networkId = Number(configService.env.NETWORK);
+
+export type TransactionsMap = Record<string, Transaction>;
+
+export type TransactionState = {
+  [networkId: number]: TransactionsMap;
+};
+
+const PERSIST_TRANSACTIONS = false;
+
+// TODO: What happens if the structure changes? Either keep a version or schema validator.
+export const transactionsState = ref<TransactionState>(
+  PERSIST_TRANSACTIONS ? lsGet<TransactionState>(LS_KEYS.Transactions, {}) : {}
+);
+
+// COMPUTED
+const transactions = computed(() =>
+  orderBy(Object.values(getTransactions()), 'addedTime', 'desc').filter(
+    isTransactionRecent
+  )
+);
+
+const pendingTransactions = computed(() =>
+  transactions.value.filter(({ status }) => status === 'pending')
+);
+
+const pendingOrderActivity = computed(() =>
+  pendingTransactions.value.filter(({ type }) => type === 'order')
+);
+
+const pendingTxActivity = computed(() =>
+  pendingTransactions.value.filter(({ type }) => type === 'tx')
+);
+
+// METHODS
+function normalizeTxReceipt(receipt: TransactionReceipt) {
+  return {
+    blockHash: receipt.blockHash,
+    blockNumber: receipt.blockNumber,
+    contractAddress: receipt.contractAddress,
+    from: receipt.from,
+    status: receipt.status,
+    to: receipt.to,
+    transactionHash: receipt.transactionHash,
+    transactionIndex: receipt.transactionIndex
+  };
+}
+
+function isTransactionRecent(transaction: Transaction): boolean {
+  return Date.now() - transaction.addedTime < DAY_MS;
+}
+
+function clearAllTransactions() {
+  setTransactions({});
+}
+
+function getId(id: string, type: TransactionType) {
+  return `${type}_${id}`;
+}
+
+function getTransactions(): TransactionsMap {
+  const transactionsMap = transactionsState.value[networkId] ?? {};
+
+  return transactionsMap;
+}
+
+function setTransactions(transactionsMap: TransactionsMap) {
+  transactionsState.value[networkId] = transactionsMap;
+  console.log(transactionsMap);
+  if (PERSIST_TRANSACTIONS) {
+    lsSet(LS_KEYS.Transactions, transactionsState.value);
+  }
+}
+
+function getTransaction(id: string, type: TransactionType) {
+  const transactionsMap = getTransactions();
+  const txId = getId(id, type);
+
+  return transactionsMap[txId];
+}
+
+function updateTransaction(
+  id: string,
+  type: TransactionType,
+  updates: Partial<Transaction>
+) {
+  const transactionsMap = getTransactions();
+  const txId = getId(id, type);
+  const transaction = transactionsMap[txId];
+
+  if (transaction != null) {
+    transactionsMap[txId] = merge(transaction, updates);
+
+    setTransactions(transactionsMap);
+
+    return true;
+  }
+
+  return false;
+}
+
+// Adapted from Uniswap code
+function shouldCheckTx(transaction: Transaction, lastBlockNumber: number) {
+  if (transaction.status === 'confirmed') {
```

Comment:
By storing the blocknumber it's possible to make these requests a bit more efficient if they are stuck. I'll add a similar code for Gnosis order handling. I don't think it's really efficient to check it every block if it's kinda stuck...